File: Server/models/blog.js
const mongoose = require("mongoose");

const blogSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: true,
    },
    content: {
      type: String,
      required: true,
    },
    tags: [
      {
        type: mongoose.Types.ObjectId,
        ref: "Tag",
      },
    ],
    likes: [
      {
        type: mongoose.Types.ObjectId,
        ref: "User",
      },
    ],
    dislikes: [
      {
        type: mongoose.Types.ObjectId,
        ref: "User",
      },
    ],
    image: {
      type: String,
    },
    author: {
      type: mongoose.Types.ObjectId,
      ref: "User",
    },
    comments: [
      {
        type: mongoose.Types.ObjectId,
        ref: "Comment",
      },
    ],
  },
  {
    timestamps: true,
  }
);

module.exports = mongoose.model("Blog", blogSchema);

File: blog/src/main/java/com/mimingucci/blog/domain/model/Blog.java
package com.mimingucci.blog.domain.model;

import lombok.Data;

import java.time.Instant;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Data
public class Blog {
    private Long id;

    private String title;

    private String content;

    private List<String> tags;

    private Long author;

    private Set<Long> likes = new HashSet<>();

    private Set<Long> dislikes = new HashSet<>();

    private Instant createdAt;

    private Instant updatedAt;
}
